training_engine_sae.py

```diff
@@ -0,0 +1,152 @@
+import cv2
+import numpy as np
+import random as rd
+from keras.models import Model
+from keras.layers import Dropout, UpSampling2D, Concatenate
+from keras.layers import Conv2D, MaxPooling2D, Input
+from keras.optimizers import Adam
+from keras.callbacks import EarlyStopping,ModelCheckpoint
+from keras.backend import image_data_format
+
+# ===========================
+#       SETTINGS
+# ===========================
+
+VALIDATION_SPLIT=0.2
```

Comment:
Does 80/20 for training/validations work nicely always? Or shall we expose this value as a setting to tune by the user with a default of 0.2?

Reply:
I don't think it is necessary to expose this value as a parameter to the user. 80/20 is rather general.